cli.go

```diff
@@ -0,0 +1,502 @@
+package jwtauth
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/signal"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/hashicorp/vault/api"
+)
+
+const defaultMount = "oidc"
+const defaultPort = "8300"
+
+type CLIHandler struct{}
+
+type loginResp struct {
+	secret *api.Secret
+	err    error
+}
+
+func (h *CLIHandler) Auth(c *api.Client, m map[string]string) (*api.Secret, error) {
+	// handle ctrl-c while waiting for the callback
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch, os.Interrupt)
+	defer signal.Stop(ch)
+
+	doneCh := make(chan loginResp)
+
+	mount, ok := m["mount"]
+	if !ok {
+		mount = defaultMount
+	}
+
+	port, ok := m["port"]
+	if !ok {
+		port = defaultPort
+	}
+
+	role := m["role"]
+	if role == "" {
+		return nil, errors.New("a 'role' must be specified")
+	}
+
+	secret, err := fetchAuthURL(c, role, mount, port)
```

Comment:
Should we name this something else? The naming of this method seems slightly confusing to me because it's returning more than an auth URL, and also because below an auth URL is checked for and there's an error if it doesn't exist, which I would expect to happen in _this_ method given its name.

Reply:
I've refactored things so the empty check now occurs in the `fetchAuthURL` method, returning an error if it fails. That should hopefully improve the consistency with the name.